table/block_based/block_based_table_iterator.cc

```diff
@@ -79,13 +79,20 @@ void BlockBasedTableIterator::SeekImpl(const Slice* target,
     }
   }
 
-  if (read_options_.auto_readahead_size && read_options_.iterate_upper_bound) {
+  if (read_options_.auto_readahead_size && read_options_.iterate_upper_bound &&
+      is_first_pass) {
```

Comment:
This change is for async_io. We need to calculate the readhead only for first pass during Seek.